utils/exec.go

```diff
@@ -135,3 +147,97 @@ func RunCmd(command string) (output string, exitCode int, startedAt time.Time, f
 	log.Infof("Exit Code: %d", exitCode)
 	return
 }
+
+func RunContinuousReportRun(fn func(chunk string) error, cmdArg string, thresholds Thresholds) (int, error) {
+	log.Debug("RunContinuousReportRun")
+
+	// command thresholds flags
+	if !(thresholds.Lines > 0) {
+		thresholds.Lines = DefaultThresholdLines
+	}
+	if !(thresholds.Time > 0) {
+		thresholds.Time = DefaultThresholdTime
+	}
+	if !(thresholds.Bytes > 0) {
+		thresholds.Bytes = DefaultThresholdBytes
+	}
+	log.Debug("Threshold", thresholds)
+
+	// Saves script/command in a temp file
+	cmdFileName := strings.Join([]string{time.Now().Format(TimeLayoutYYYYMMDDHHMMSS), "_", RandomString(10)}, "")
+	if runtime.GOOS == "windows" {
+		cmdFileName = strings.Join([]string{cmdFileName, ".bat"}, "")
+	}
+
+	// Writes content
+	if err := ioutil.WriteFile(cmdFileName, []byte(cmdArg), 0600); err != nil {
+		log.Fatalf("Error creating temp file : ", err)
+	}
+
+	// Creates command
+	var newCmd *exec.Cmd
+	if runtime.GOOS == "windows" {
+		newCmd = exec.Command("cmd", "/C", cmdFileName)
+	} else {
+		newCmd = exec.Command("/bin/sh", cmdFileName)
+	}
+
+	// Removes temp file
+	defer func() {
+		err := os.Remove(cmdFileName)
+		if err != nil {
+			log.Warn("Temp file cannot be removed", err.Error())
+		}
+	}()
+
+	// Gets the pipe command
+	stdout, err := newCmd.StdoutPipe()
+	if err != nil {
+		return 1, fmt.Errorf("cannot get pipe command %v", err)
+	}
+	log.Info("==> Executing: ", strings.Join(newCmd.Args, " "))
+
+	// Start command asynchronously
+	if err = newCmd.Start(); err != nil {
+		return 1, fmt.Errorf("cannot start the specified command %v", err)
+	}
+
+	chunk := ""
+	nLines, nTime, nBytes := 0, 0, 0
+	timeStart := time.Now()
+
+	scanner := bufio.NewScanner(bufio.NewReader(stdout))
+	for scanner.Scan() {
+		chunk = strings.Join([]string{chunk, scanner.Text(), "\n"}, "")
+		nLines++
+		nTime = int(time.Now().Sub(timeStart).Seconds())
+		nBytes = len(chunk)
+		if nLines >= thresholds.Lines || nTime >= thresholds.Time || nBytes >= thresholds.Bytes {
```

Comment:
This is working great, but after thoughtful consideration I think the only parameters we should be considering is time and whether there are any pending lines to be sent. If we force it to send lines once every 5 seconds (and only if there are pending lines) we are not losing much responsiveness but we will be saving greatly in API requests. Regarding the number of bytes, I actually think that the larger the request we send, the better because it makes the process more efficient.

Reply:
So, if i understand fine, should be managed only by time? thus ,lines/bytes counter has to be removed